### ISSUE TITLE
设置user_inter_num_interval = [5, inf)后得到的序列长度仍然有小于5的序列

### ISSUE THREAD

**Issue by lalunex** [RUCAIBox/RecBole#1585]
您好，我遇到的问题如下图所示：
![image](https://user-images.githubusercontent.com/63094854/207595054-250baa30-5827-4152-affe-915960699e5e.png)

下面是参数：
General Hyper Parameters:
gpu_id = 0
use_gpu = True
seed = 42
state = INFO
reproducibility = True
data_path = E:\anaconda3\envs\ll_torch10\Lib\site-packages\recbole\config\../dataset_example/ml-100k
checkpoint_dir = saved
show_progress = True
save_dataset = False
dataset_save_path = None
save_dataloaders = False
dataloaders_save_path = None
log_wandb = False
Training Hyper Parameters:
epochs = 300
train_batch_size = 256
learner = adam
learning_rate = 0.001
neg_sampling = {'uniform': 1}
eval_step = 1
stopping_step = 10
clip_grad_norm = {'max_norm': 5.0, 'norm_type': 2}
weight_decay = 0
loss_decimal_place = 4
Evaluation Hyper Parameters:
eval_args = {'group_by': 'user', 'order': 'TO', 'split': {'LS': 'valid_and_test'}, 'mode': 'full'}
repeatable = True
metrics = ['Hit', 'NDCG', 'MRR']
topk = [1, 5, 10, 20]
valid_metric = Hit@20
valid_metric_bigger = True
eval_batch_size = 4096
metric_decimal_place = 4
Dataset Hyper Parameters:
field_separator = 	
seq_separator =  
USER_ID_FIELD = user_id
ITEM_ID_FIELD = item_id
RATING_FIELD = rating
TIME_FIELD = timestamp
seq_len = None
LABEL_FIELD = label
threshold = None
NEG_PREFIX = neg_
load_col = {'inter': ['user_id', 'item_id', 'rating', 'timestamp']}
unload_col = None
unused_col = None
additional_feat_suffix = None
rm_dup_inter = None
val_interval = None
filter_inter_by_user_or_item = True
user_inter_num_interval = [5, inf)
item_inter_num_interval = [5, inf)
alias_of_user_id = None
alias_of_item_id = None
alias_of_entity_id = None
alias_of_relation_id = None
preload_weight = None
normalize_field = None
normalize_all = None
ITEM_LIST_LENGTH_FIELD = item_length
LIST_SUFFIX = _list
MAX_ITEM_LIST_LENGTH = 50
POSITION_FIELD = position_id
HEAD_ENTITY_ID_FIELD = head_id
TAIL_ENTITY_ID_FIELD = tail_id
RELATION_ID_FIELD = relation_id
ENTITY_ID_FIELD = entity_id
benchmark_filename = None
Other Hyper Parameters: 
wandb_project = recbole
require_pow = False
MODEL_TYPE = ModelType.SEQUENTIAL
scheduler = False
step_size = 4
gamma = 0.1
num_layers = 1
dropout_prob = 0.3
loss_type = BPR
initializer_range = 0.02
step = 1
sub_model = BERT4Rec
data_augmentation = True
our_att_drop_out = 0.2
our_ae_drop_out = 0.3
gumbel_temperature = 0.5
is_gumbel_tau_anneal = True
long_tail_rate = 0.95
train_neg_sample_args = {'strategy': 'by', 'by': 1, 'distribution': 'uniform', 'dynamic': 'none'}
pre_train_model_dict = {'amazon-sports-outdoors': {'GRU4Rec': 'saved/GRU4Rec-May-17-2022_13-14-44.pth', 'NARM': 'saved/NARM-May-17-2022_13-16-07.pth', 'Caser': 'saved/Caser-May-17-2022_13-17-42.pth', 'BERT4Rec': 'saved/BERT4Rec-May-17-2022_13-20-26.pth', 'SASRec': 'saved/SASRec-May-17-2022_13-27-07.pth', 'STAMP': 'saved/STAMP-May-17-2022_14-20-57.pth', 'DSAN': 'saved/DSAN-May-19-2022_23-25-56.pth'}, 'ml-100k': {'GRU4Rec': 'saved/GRU4Rec-May-17-2022_13-13-42.pth', 'NARM': 'saved/NARM-May-19-2022_19-31-33.pth', 'Caser': 'saved/Caser-May-17-2022_13-13-56.pth', 'BERT4Rec': 'saved/BERT4Rec-May-17-2022_13-14-04.pth', 'SASRec': 'saved/SASRec-May-17-2022_13-14-14.pth', 'STAMP': 'saved/STAMP-May-17-2022_13-14-49.pth'}, 'yelp': {'GRU4Rec': 'saved/GRU4Rec-May-17-2022_13-15-01.pth', 'NARM': 'saved/NARM-May-17-2022_13-17-59.pth', 'Caser': 'saved/Caser-May-17-2022_13-26-00.pth', 'BERT4Rec': 'saved/BERT4Rec-May-17-2022_13-30-05.pth', 'SASRec': 'saved/SASRec-May-17-2022_13-45-49.pth', 'STAMP': 'saved/STAMP-May-17-2022_14-27-11.pth'}, 'amazon-beauty': {'GRU4Rec': 'saved/GRU4Rec-May-17-2022_13-13-26.pth', 'NARM': 'saved/NARM-May-17-2022_13-14-20.pth', 'Caser': 'saved/Caser-May-17-2022_13-16-13.pth', 'BERT4Rec': 'saved/BERT4Rec-May-17-2022_13-17-49.pth', 'SASRec': 'saved/SASRec-May-17-2022_13-21-48.pth', 'STAMP': 'saved/STAMP-May-17-2022_14-05-50.pth'}, 'amazon-clothing-shoes-jewelry': {'GRU4Rec': 'saved/GRU4Rec-May-17-2022_13-32-16.pth', 'NARM': 'saved/NARM-May-17-2022_13-34-29.pth', 'Caser': 'saved/Caser-May-17-2022_13-37-12.pth', 'BERT4Rec': 'saved/BERT4Rec-May-17-2022_13-40-44.pth', 'SASRec': 'saved/SASRec-May-17-2022_13-46-02.pth', 'STAMP': 'saved/STAMP-May-17-2022_15-03-23.pth'}, 'ml-1m': {'GRU4Rec': 'saved/GRU4Rec-May-17-2022_13-13-41.pth', 'NARM': 'saved/NARM-May-17-2022_13-14-36.pth', 'Caser': 'saved/Caser-May-17-2022_13-15-33.pth', 'BERT4Rec': 'saved/BERT4Rec-May-17-2022_13-16-38.pth', 'SASRec': 'saved/SASRec-May-17-2022_13-27-12.pth', 'STAMP': 'saved/STAMP-May-17-2022_13-39-43.pth'}}
load_pre_train_emb = False
user_short_nh = 8
user_short_nv = 4
sequence_last_m = 5
embedding_size = 100
n_layers = 2
n_heads = 2
hidden_size = 100
inner_size = 100
hidden_dropout_prob = 0.5
attn_dropout_prob = 0.5
hidden_act = gelu
layer_norm_eps = 1e-12
mask_ratio = 0.6
MODEL_INPUT_TYPE = InputType.PAIRWISE
eval_type = EvaluatorType.RANKING
device = cuda
eval_neg_sample_args = {'strategy': 'full', 'distribution': 'uniform'}

**Comment by Paitesanshi** [comment]
@lalunex 您好，感谢您对伯乐的关注！
设置user_inter_num_interval = [5, inf)后保证用户的总交互数大于5，但是用户的交互序列会分散在训练接和测试集中，不能保证train/valid/test data中的seq_len都大于5。

**Comment by Paitesanshi** [comment]
@lalunex 您设置的划分方式为valid_and_test, 那么valid data和test data中长度应该是1，train_data中为剩下的交互

**Comment by lalunex** [comment]
@Paitesanshi 您好，请问那么有没有办法可以使train data里面的序列长度大于等于5，且valid data和test data中长度是1的方法呢？

**Comment by Paitesanshi** [comment]
@lalunex 您好，可以尝试修改设置user_inter_num_interval = [7, inf)，并继续使用valid_and_test划分

**Comment by lalunex** [comment]
@Paitesanshi 好的，谢谢，叨扰了
